feat(eslint-plugin): [no-confusing-void-expression] add an option to ignore void<->void (#10067)
typescript-eslint/typescript-eslint: packages/eslint-plugin/docs/rules/no-confusing-void-expression.mdx

console.log(void alert('Hello, world!'));
```

### `ignoreVoidReturningFunctions`

> Whether to ignore returns from functions with explicit `void` return types and functions with contextual `void` return types.

Reviewer: [Docs] I think most users won't understand what _"contextual"_  means. How does this strike you?

```suggestion
Whether to ignore returns from functions with `void` return types when inside a function with a `void` return type.
```

Author: Yes! This is a lot better, thanks ❤️

> It may be preferable to allow functions that explicitly return `void` to return `void` expressions. This exception will also apply to functions with an implicit contextual return type of `void`.

Reviewer: [Docs] It's good to be explicit on _why_ something is or isn't preferable:

```suggestion
Some projects prefer allowing functions that explicitly return `void` to return `void` expressions. Doing so allows more writing more succinct functions.

:::note
This is technically risky as the `void`-returning function might actually be returning a value not seen by the type system.
:::
```

Author: Can't agree more on this, thanks 👍
